app/services/plans/override_service.rb

```diff
@@ -36,6 +36,12 @@ def call
           Charges::OverrideService.call(charge:, params: charge_params)
         end
 
+        if params[:minimum_commitment].present? && License.premium?
+          commitment = Commitment.new(plan: new_plan, commitment_type: 'minimum_commitment')
```

Comment:
The logic is a bit different in case of commitment - when overriding a plan we're creating a new commitment that belongs to this overriden plan. The commitment that belong to the original plan staus intact.